src/java/org/apache/cassandra/index/sai/plan/Plan.java

```diff
@@ -1764,7 +1813,7 @@ public TableMetrics(long rows, double avgCellsPerRow, double avgBytesPerRow, int
     public interface Executor
     {
         Iterator<? extends PrimaryKey> getKeysFromIndex(Expression predicate);
-        Iterator<? extends PrimaryKey> getTopKRows(int softLimit);
+        Iterator<? extends PrimaryKey> getTopKRows(Expression predicate, int softLimit);
         Iterator<? extends PrimaryKey> getTopKRows(RangeIterator keys, int softLimit);
```

Comment:
When we have intersection between columns `a` and `b` and then order on `b`, does it do the right thing here and either get the slice from the ordered index or filter both a/b and then sort the results?

Reply:
Also, we could conceivably have `a < 1 OR a > 10 ORDER BY a` where we union two ordered iterators. Not sure if we want to handle that here or in another PR.

Reply:
Yes, that is the idea.